src/stdio_mgr/stdio_mgr.py

```diff
@@ -72,10 +77,13 @@ def write(self, *args, **kwargs):
         super().write(*args, **kwargs)
         self.flush()
 
+    def _set_closed_buf(self, value):
+        self._closed_buf = value
+
     def getvalue(self):
         """Obtain buffer of text sent to the stream."""
         if self._stream.closed:
-            return self._stream._closed_buf.decode(self.encoding)
+            return self._closed_buf.decode(self.encoding)
```

Comment:
Yep, returned as `str`.

And, this seems like it's better, too, b/c it's not reaching down into an attribute on an internal private object to access the stored value.

Reply:
This is documented as emulating `StringIO.getvalue` , which it replaced , not `BytesIO.getvalue`